athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/filter/FilterExpressionBuilder.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.amazonaws.athena.connector.lambda.domain.predicate.Marker.Bound.EXACTLY;
+
+/**
+ * This class is used to build Filter Expression to handle the constraints on partition folder
+ *
+ */
+public class FilterExpressionBuilder
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(FilterExpressionBuilder.class);
+
+    /**
+     * List of {@link Field} instances
+     */
+    private final List<Field> fields;
+
+    private final List<EqualsExpression> equalsExpressions = new ArrayList<>();
+
+    public FilterExpressionBuilder(Schema schema)
+    {
+        this.fields = schema.getFields();
+    }
+
+    /**
+     * Prepares the expressions based on the provided arguments.
+     *
+     * @param constraints            An instance of {@link Constraints} that is a summary of where clauses (if any)
+     * @return A list of {@link EqualsExpression}
+     */
+    public List<EqualsExpression> getExpressions(Constraints constraints)
+    {
+        LOGGER.info("Constraint summaries: \n{}", constraints.getSummary());
+        List<EqualsExpression> conjuncts = new ArrayList<>();
+        for (Field column : fields) {
+            if (constraints.getSummary() != null && !constraints.getSummary().isEmpty()) {
+                ValueSet valueSet = constraints.getSummary().get(column.getName());
+                LOGGER.info("Value set for column {} was {}", column, valueSet);
+                if (valueSet != null) {
+                    conjuncts.addAll(addFilterExpressions(column.getName(), valueSet));
+                }
+            }
+        }
+        return conjuncts;
+    }
+
+    /**
+     * Add one or more {@link EqualsExpression} based on {@link ValueSet}
+     *
+     * @param columnName The name of the column
+     * @param valueSet   An instance of {@link ValueSet}
+     * @return List of {@link EqualsExpression}
+     */
+    private List<EqualsExpression> addFilterExpressions(String columnName, ValueSet valueSet)
+    {
+        LOGGER.info("FilterExpressionBuilder::addFilterExpressions -> Evaluating and adding expression for col {} with valueSet {}", columnName, valueSet);
+        List<EqualsExpression> disjuncts = new ArrayList<>();
+        List<Object> singleValues = new ArrayList<>();
+        LOGGER.info("FilterExpressionBuilder::addFilterExpressions -> Bound {}", valueSet.getRanges().getSpan().getLow().getBound());
+        if (!valueSet.isNullAllowed() && isHighLowEquals(valueSet)
+                && valueSet.getRanges().getSpan().getLow().getBound().equals(EXACTLY)) {
+            return List.of(new EqualsExpression(columnName.toLowerCase(), valueSet.getRanges().getSpan().getLow().getValue()));
+        }
+
+        for (Range range : valueSet.getRanges().getOrderedRanges()) {
+            if (range.isSingleValue()) {
+                singleValues.add(range.getLow().getValue());
+            }
+        }
+        if (singleValues.size() == 1) {
```

Comment:
What does it mean if `singleValues.size()` is not 1?

If this is only ever expected to be 1 or 0, then can't we just add to `equalsExpressions` directly? As in, what is the purpose of `singleValues` and `disjuncts` ?

Additionally, I noticed that in the other return path above (line 92), you are not adding anything into `equalsExpressions`, is that intentional?

Which leads me to the other question of whether or not the member variable `equalsExpressions` is even necessary if the other return path is just returning the `List` without adding it to `equalsExpressions`?